libtextsecure/account_manager.js

```diff
@@ -517,8 +523,8 @@
       });
     },
     async generateMnemonic(language = 'english') {
-      const keys = await libsignal.KeyHelper.generateIdentityKeyPair();
-      const hex = StringView.arrayBufferToHex(keys.privKey);
+      const seed = window.Signal.Crypto.getRandomBytes(16);
```

Comment:
I don't fully understand this change, what was generateIdentityKeyPair doing and why can we just get random bytes instead?

Reply:
Right, apologies for the lack of context.

During the registration step, the "register a new account" tab comes with a pre-defined seed word list, which is the only use of the function `generateMnemonic`.

Then, when proceeding to the actual account creation, the private key is always derived from the mnemonic seed, regardless of whether it was manually provided (via "restore using seed") or using the pre-generated seed returned by `generateMnemonic`.

So basically, I used to use `generateIdentityKeyPair` as a 32 bytes seed generator, which was confusing (seed != private key).

To show the difference in a flow chart (when using the "register a new account" path):

before:
`generate 32bytes private key => convert to 24 mnemonic words => convert to 32bytes seed => convert to 32bytes private key`

now:
`generate random 16bytes seed => convert to 12 mnemonic words => convert to 16bytes seed => extend to 32bytes by duplication => convert to 32bytes private key`

Reply:
Makes sense to me!